ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
```

Comment:
pthread-rwlock isn't re-entrant, is it? `getItemsWithKey:userId:correlationId:error:` calls `getItemsImplKey:userId:`.